### ISSUE TITLE
[0.7.1] default priority systems can be scheduled after systems with higher priority

### ISSUE THREAD

**Issue by benbrimeyer** [matter-ecs/matter#48]
## Description
When setting a system's priority higher than 0, the expectation is that it will be scheduled only after the default priority systems. In 0.7.0, this appeared to be the case, but in 0.7.1 we can observe that an `after` can "pull down" a system to a lower priority in the queue.

## Steps to reproduce

1. Download the attached zip file for a small matter project made to reproduce this behavior. [matter-repro-systems.zip](https://github.com/matter-ecs/matter/files/14513503/matter-repro-systems.zip)
2. Build/serve it to a default studio baseplate.
3. Press Play Solo.
4. Observe the output.

## Expected behavior

There are three systems in this project, `a`, `b`, and `last`. The output will print the following message describing my expectations:
```
expectation:
	system b            (default priority)
	system last         (1000 priority)
	system a            (after last)
```
## Actual behavior
On 0.7.1, we get this schedule instead:
```
system last
system a
system b
```

This is problematic because `system b` has a default priority and is being scheduled after `system last` which has a priority of 1000. My guess is that `system a`'s `after` field is "pulling down" `system last` in the schedule...

Using 0.7.0 will run the schedule as expected.

**Comment by benbrimeyer** [comment]
its likely this PR introduced this change in behavior: https://github.com/matter-ecs/matter/pull/1
